pkg/services/repo_service.go

```diff
@@ -141,9 +140,9 @@ func (a *argoCDService) GetFileContent(ctx context.Context, repoURL string, revi
 		return nil, err
 	}
 
-	bytes, err := ioutil.ReadFile(filepath.Join(gitRepoClient.Root(), path))
+	bytes, err := os.ReadFile(filepath.Join(gitRepoClient.Root(), path))
 	if err != nil {
-		return nil, err
+		return nil, err.(*os.PathError).Err
```

Comment:
Can we return the error without casting it to (`*os.PathError`)? This seems like it could break us if the underlying implementation were to change. 

What if we just `return nil, err` here as before? Any issues there?

Reply:
I dont think if that is true, its core lib. Its same as we are dealing with any logic. But i am fine with both. 

Let me replace it with err , but still we are comparing the error strings and it could change anytime so it will break the things.

So in both way, expectations are same. But i like the idea of checking the path inside that string so will just revert it back what it was with some fix in the test itself for correct path generation.